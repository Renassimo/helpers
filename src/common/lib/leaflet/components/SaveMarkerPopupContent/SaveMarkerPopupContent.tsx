import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const SaveMarkerPopupContent = ({
  onAdd,
  onCancel,
  text = 'Add item to map?',
}: {
  onAdd: () => void;
  onCancel: () => void;
  text?: string;
}) => {
  return (
    <Box>
      <Typography>{text}</Typography>
      <Box>
        <Button
          size="small"
          onClick={(event) => {
            event.stopPropagation();
            onAdd();
          }}
        >
          Yes
        </Button>
        <Button
          size="small"
          onClick={(event) => {
            event.stopPropagation();
            onCancel();
          }}
          color="error"
        >
          No
        </Button>
      </Box>
    </Box>
  );
};

export default SaveMarkerPopupContent;
