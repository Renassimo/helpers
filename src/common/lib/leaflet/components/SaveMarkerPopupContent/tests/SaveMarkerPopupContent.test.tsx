import renderWithTheme from '@/common/tests/helpers/renderWithTheme';
import SaveMarkerPopupContent from '../SaveMarkerPopupContent';
import userEvent from '@testing-library/user-event';

describe('SaveMarkerPopupContent', () => {
  const mockedOnAdd = jest.fn();
  const mockedOnCancel = jest.fn();

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('matches snapshot', () => {
    // Arange
    // Act
    const { baseElement } = renderWithTheme(
      <SaveMarkerPopupContent onAdd={mockedOnAdd} onCancel={mockedOnCancel} />
    );
    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  describe('when tesxt passed', () => {
    test('matches snapshot', () => {
      // Arange
      // Act
      const { baseElement } = renderWithTheme(
        <SaveMarkerPopupContent
          onAdd={mockedOnAdd}
          onCancel={mockedOnCancel}
          text="mocked-text"
        />
      );
      // Assert
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('when onAdd clicked', () => {
    test('matches snapshot', async () => {
      // Arange
      const { getByText } = renderWithTheme(
        <SaveMarkerPopupContent
          onAdd={mockedOnAdd}
          onCancel={mockedOnCancel}
          text="mocked-text"
        />
      );
      // Act
      await userEvent.click(getByText('Yes'));
      // Assert
      expect(mockedOnAdd).toHaveBeenCalledTimes(1);
      expect(mockedOnCancel).not.toHaveBeenCalled();
    });
  });

  describe('when onCancel clicked', () => {
    test('matches snapshot', async () => {
      // Arange
      const { getByText } = renderWithTheme(
        <SaveMarkerPopupContent
          onAdd={mockedOnAdd}
          onCancel={mockedOnCancel}
          text="mocked-text"
        />
      );
      // Act
      await userEvent.click(getByText('No'));
      // Assert
      expect(mockedOnAdd).not.toHaveBeenCalled();
      expect(mockedOnCancel).toHaveBeenCalledTimes(1);
    });
  });
});
