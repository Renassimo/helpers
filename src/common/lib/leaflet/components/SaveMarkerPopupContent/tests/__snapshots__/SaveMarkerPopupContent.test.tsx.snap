// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveMarkerPopupContent matches snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Add item to map?
      </p>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Yes
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError css-6ekzlo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          No
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`SaveMarkerPopupContent when tesxt passed matches snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        mocked-text
      </p>
      <div
        class="MuiBox-root css-0"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Yes
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError MuiButton-root MuiButton-text MuiButton-textError MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorError css-6ekzlo-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          No
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;
